### PR TITLE
Add "Publishing Modules" section

### LINKED ISSUE #191: Add section on distributing Deno modules for use in Node (dnt)
We really need to add section on this to the manual.

### PR DESCRIPTION
**Pull request by dsherret**
Took text from other parts of the site for this.

Closes #191

### CODE REVIEW

File: toc.json [hunk 1]
```diff
@@ -132,6 +132,9 @@
       "imports": "Import suggestions and intelligent registries"
     }
   },
+  "publishing": {
+    "name": "Publishing Modules"
+  },
```
Comment (ry) [inline toc.json:137]: I feel like there should be a section in the "Using npm/Node.js code" chapter. 
"How to publish to NPM"

Comment (dsherret) [reply toc.json:138]: Added a "dnt" page under there.

File: toc.json [hunk 2]
```diff
@@ -47,7 +47,8 @@
       "cdns": "Packages from CDNs",
       "import_maps": "Using import maps",
       "faqs": "Frequently asked questions",
-      "cheatsheet": "Node->Deno cheatsheet"
+      "cheatsheet": "Node->Deno cheatsheet",
+      "dnt": "dnt"
```
Comment (dsherret) [inline toc.json:51]: Better title? I couldn't think of anything short.

Comment (kitsonk) [reply toc.json:51]: `dnt - Deno to Node Transform`?  I think it should be something more than just "dnt" because almost everyone would be huh?

**Review by kitsonk — APPROVED** [review]
We really should have this in the manual, even if we improve upon it later.

**Review by UltiRequiem — APPROVED** [review]
LGTM

**Pull request merged**